packages/contracts/contracts/L1/Colosseum.sol

```diff
@@ -642,6 +750,57 @@ contract Colosseum is Initializable, Semver {
         );
     }
 
+    /**
+     * @notice Cancels the challenge if the output root to be challenged has already been deleted.
+     *         If the output root has been deleted, delete the challenge and refund the challenger's pending bond.
+     *
+     * @param _outputIndex Index of the L2 checkpoint output.
+     * @param _challenger  Address of the challenger.
+     * @param _status      Current status of the challenge.
+     *
+     * @return Whether the challenge was canceled.
+     */
+    function _cancelIfOutputDeleted(
+        uint256 _outputIndex,
+        address _challenger,
+        ChallengeStatus _status
+    ) private returns (bool) {
+        bytes32 outputRoot = L2_ORACLE.getL2Output(_outputIndex).outputRoot;
+        if (outputRoot != DELETED_OUTPUT_ROOT) {
+            return false;
+        }
+
+        require(
+            _status != ChallengeStatus.CHALLENGER_TIMEOUT,
+            "Colosseum: challenge cannot be cancelled if challenger timed out"
+        );
+
+        // If the output is deleted, the asserter does not need to do anything further.
+        require(msg.sender == _challenger, "Colosseum: only the challenger can cancel");
```

Comment:
Could you move up this a little bit so that if the error occurs, the gas usage can be reduced.

Reply:
It can't move up because this function is also called by asserter.